sklearn/cluster/tests/test_k_means.py

```diff
@@ -1276,3 +1277,58 @@ def test_predict_does_not_change_cluster_centers(is_sparse):
 
     y_pred2 = kmeans.predict(X)
     assert_array_equal(y_pred1, y_pred2)
+
+
+@pytest.mark.parametrize("init", ["k-means++", "random"])
+def test_sample_weight_init(init, global_random_seed):
+    """Check that sample weight is used during init."""
+    rng = np.random.RandomState(global_random_seed)
+    X, _ = make_blobs(
+        n_samples=200, n_features=10, centers=10, random_state=global_random_seed
+    )
+    sample_weight = rng.uniform(size=X.shape[0])
+    x_squared_norms = row_norms(X, squared=True)
+    kmeans = KMeans(random_state=global_random_seed, init=init)
```

Comment:
I am wondering if we should parametrize on `KMeans` and `MiniBatchKMeans` as well.

Reply:
Oh I see the proposal from @jeremiedbb. Could add in the docstring that we don't need to test both estimator because they share the `_init_centroids` method. It would avoid futher questions.